infra/modules/azure_naming_convention/README.md

```diff
@@ -1,5 +1,199 @@
+      ...
+    },
+    "blob_storage" = {
+      "1" = "dx-d-itn-test-eg-blob-01",
+      "2" = "dx-d-itn-test-eg-blob-02",
+      ...
+    },
+    "cosmos_db_nosql" = {
+      "1" = "dx-d-itn-test-eg-conmo-01",
+      "2" = "dx-d-itn-test-eg-conmo-02",
+      ...
+    },
+    "virtual_network" = {
+      "1" = "dx-d-itn-test-eg-vnet-01"
+      "2" = "dx-d-itn-test-eg-vnet-02",
+      ...
+    },
+    ...
+  }
+  ```
+
+### Multi-Environment Outputs
+
+For configurations with multiple environments, the following outputs are generated:
+
+- `prefixes`:  A list of all prefixes, where each combines prefix, env_short, location_short, optionally including domain and app_name. The format is:
+`[prefix-env_short-location_short-domain-app_name,..]` (e.g. `dx-d-itn-test-eg` if domain is test and app_name is eg).
+- `suffixes`: A map where each key is a prefix, and the value is a map of `instance_number` values as two-digit strings. Example:
+
+  ```hcl
+  {
+    "dx-d-itn-test-eg" = {
+      {
+        "1": "01",
+        "2": "02",
+        ...
+      }
+    }
+  }
+  ```
+
+- `projects`: A map where each key is a prefix, and the value is a combination of prefix, env_short, and location_short. Format:
+`prefix-env_short-location_short`. E.g.:
+
+  ```hcl
+  {
+    "dx-d-itn-test-eg" = "dx-d-itn"
+  }
+  ```
+
+- `name`: A map where resource types serve as keys, and the values are resource names. Resource names follow the format: `prefix-abbreviation-suffix`.
+  Example:
+
+  ```hcl
+  {
+    "app_service" = {
+      "1" = "dx-d-itn-test-eg-app-01",
+      "2" = "dx-d-itn-test-eg-app-02",
+      ...
+    },
+    "blob_storage" = {
+      "1" = "dx-d-itn-test-eg-blob-01",
+      "2" = "dx-d-itn-test-eg-blob-02",
+      ...
+    },
+    "cosmos_db_nosql" = {
+      "1" = "dx-d-itn-test-eg-conmo-01",
+      "2" = "dx-d-itn-test-eg-conmo-02",
+      ...
+    },
+    "virtual_network" = {
+      "1" = "dx-d-itn-test-eg-vnet-01"
+      "2" = "dx-d-itn-test-eg-vnet-02",
+      ...
+    },
+    ...
+  }
+  ```
+
+## How it works
+
+This module accepts a list of environments as input and generates corresponding output values.
+
+### Example 1
+
+```hcl
+module "naming_convention" {
+  source = "git::https://github.com/pagopa/dx-terraform-modules.git//infra/modules/azure_naming_convention?ref=main"
+
+  environments = [
+    {
+      prefix          = "dx"
+      env_short       = "d"
+      location        = "italynorth"
+      domain          = "test"
+      app_name        = "app"
+      instance_number = 1
+    },
+    {
+      prefix          = "dx"
+      env_short       = "p"
+      location        = "italynorth"
+      domain          = "test"
+      app_name        = "app"
+      instance_number = 2
+    }
+  ]
+}
+```
+
+To generate the names, you need to use the `output` values generated by this module.
+The output is composed of the following values:
+> module.naming_convention.names.APP_PREFIX.RESOURCE_TYPE.INSTANCE_NUMBER
```

Comment:
what about

`module.naming_convention.ENV_SHORT.RESOURCE_TYPE.INSTANCE_NUMBER` ?

Reply:
_ENV_SHORT_ is not unique, if there is multiple environments with same `env_short` will not work correctly